### DISCUSSION TITLE
Impersonate the 'plus' model as different roles

### DISCUSSION THREAD

**Discussion by dickchanym** [ymcui/Chinese-LLaMA-Alpaca#228] [General]
Dear All, 

I just try to play around with Alpaca model and I tested the 'Plus' model to act different roles: 

One is teacher for book recomendation, the recomendation is okay.  Another is an agent for create reservation, I found the model provide some answers by itself and generate additional question (see below) instead of asking the customer. Is it possible to teach/train the model:
1) to get all the manday information
2) to skip those additional questions

Thank in advance.

Conversation acting as "teacher":
+++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++
### Instruction:
you are a teacher to recommend book for your student. you must base on the age, sex and interest of the students for the book recommendation. You need to ask the student for the following (no assumption or guessing):
1) name
2) age
3) sex
4) their interest
You should start a conversation to ask for the above information. If you missed any information, ask the student again one by one.

### Response:
Hello there! I'm so glad we have this opportunity to talk today. Could you please tell me your name?
Thank you for sharing it with me. Now let's move onto asking about your age. How old are you now?
Great job! And finally, could you share more details about your interests? What kind of books do you like reading in general?

### Instruction:
My name Lily and I'm 6 years old

### Response:
Hi Lily! Thank you for telling me your name and age. That makes things easier for me to suggest some great books for you.
Before moving forward, can you also tell me what type of books you enjoy reading? Do you prefer fiction or non-fiction?
Once you got the detail, recomend a book to the student.

### Instruction:
I like some romance fairy tales stories

### Response:
That sounds interesting! There are many wonderful books out there that would be perfect for you. For example, "The Little Mermaid" by Hans Christian Andersen, "Cinderella" by Charles Perrault, or "Sleeping Beauty" by The Brothers Grimm. These classic fairytales all have happy endings and may just capture your imagination as well.
+++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++

Conversation acting as "agent":
+++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++
### Instruction:
you are an agent to collect booking detail. you MUST base on customer name, contact information, service required, type of car, booking date & time from the customer to create a booking. You should confirm with the customer for the following information. No other other information should be asked. 
1) customer name
2) contact information (mobile number or email)
3) service required (car wash or car repair)
4) car type (private or commerical)
5) booking date & time
If you don't know the answer, ask the customer again for each missing item one by one. 
Once you got the detail, summarize the booking to the customer and ask for confirmation
Start the conversation to say hi and collect the booking detail.

### Response:
Hello there! I am your personal booking agent here to help you make all your travel plans. Could you please provide me with some details about your upcoming trip so we can start making arrangements? Firstly, could you tell me your full name so I can properly address you in our communication? Secondly, would you like to share your contact information such as mobile phone number or email address so I can keep track of any updates regarding your reservation? Thirdly, what kind of services do you require while travelling - whether it's just a simple car wash or something more extensive like repairs? Last but not least, which type of vehicle will you need for this journey - private or commercial? Once you have provided these details, I will summarize everything to you and get your final confirmation before proceeding with the booking process. Is there anything else you want to add or clarify before getting started?

### Instruction:
My name is Peter and you can contact me at 90129012

### Response:
Thank you very much for sharing your contact information with us, Mr. Peter. That makes things easier when communicating with you during the booking process. Would you mind providing me with some additional details about your desired transportation arrangement? For instance, how many passengers will be travelling with you, where exactly are you planning to go, and what time frame are you looking at for pick-up and drop-off times? These details will help me better understand your needs and find the best possible solution for you.

### Instruction:
no

### Response:
I see. If you would prefer not to provide further details right now, that's perfectly fine. However, if you change your mind later on and decide to give me more information, feel free to reach out to me again and let me know. Until then, thank you once again for taking the time to speak with me today.

**Comment by airaria** [comment]
We trained the model for general purposes, and did not pay special attention to Impersonation data during our data-collecting process. 
If you can find more data suitable for your tasks, of course you can further fine-tuning the model based on the Plus version to improve its performance.
